### PR TITLE
Fix a PIX pass's attempt to set the validator version

### PR DESCRIPTION
**Pull request by jeffnn**
This pass was attempting to compare different things. The return values of GetDxilVersion are not shader models, but... dxil version. Since the code is trying to upgrade the validator version, I changed this to GetValidatorVersion, to pair with SetValidatorVersion.
The previous code was breaking the nvidia driver on workgraphs.

### CODE REVIEW

File: lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp [hunk 1]
```diff
@@ -123,8 +123,8 @@ bool DxilAnnotateWithVirtualRegister::runOnModule(llvm::Module &M) {
   }
   unsigned int Major = 0;
   unsigned int Minor = 0;
-  m_DM->GetDxilVersion(Major, Minor);
-  if (Major < 6 || (Major == 6 && Minor <= 4)) {
+  m_DM->GetValidatorVersion(Major, Minor);
+  if (Major < 1 || (Major == 1 && Minor < 4)) {
```
Comment (tex3d) [inline lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp:127]: If `Major < 1`, the validator version is probably 0.0, which means it's never meant to be validated, and shouldn't be changed to 1.4.  You should use `DXIL::CompareVersions` to compare, as it takes this into account.

```suggestion
  if (DXIL::CompareVersions(Major, Minor, 1, 4) < 0)) {
```

File: lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp [hunk 2]
```diff
@@ -177,7 +177,6 @@ bool DxilAnnotateWithVirtualRegister::runOnModule(llvm::Module &M) {
                   << printableNameSubset << "\n";
     }
   }
-
```
Comment (tex3d) [inline lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp:180]: nit: is this intentional?

File: tools/clang/test/DXC/Passes/PIX/validatorUpgrade.ll [hunk 3]
```diff
@@ -0,0 +1,64 @@
+; RUN: %dxopt %s -dxil-annotate-with-virtual-regs -hlsl-dxilemit -S | FileCheck %s
+
+; CHECK: !dx.valver = !{![[VALVER:.*]]}
+; CHECK-NOT: ![[VALVER]] = !{i32 1, i32 4}
```
Comment (tex3d) [inline tools/clang/test/DXC/Passes/PIX/validatorUpgrade.ll:4]: This ensures we don't downgrade the version, but is there already a test to make sure we upgrade the validator version in the appropriate scenario somewhere else?

File: lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp [hunk 4]
```diff
@@ -123,8 +123,8 @@ bool DxilAnnotateWithVirtualRegister::runOnModule(llvm::Module &M) {
   }
   unsigned int Major = 0;
   unsigned int Minor = 0;
-  m_DM->GetDxilVersion(Major, Minor);
-  if (Major < 6 || (Major == 6 && Minor <= 4)) {
+  m_DM->GetValidatorVersion(Major, Minor);
```
Comment (tex3d) [inline lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp:126]: Wha's the original reason for upgrading the validator version to 1.4 specifically?

Comment (jeffnn) [reply lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp:126]: I wish I could ask jpporto that question... this pass does nothing but add some metadata, so maybe the metadata format changed in 1.4? I also wonder if it's here on behalf of the two other passes that are typically run with this one (value-to-delcare and shader debug instrumentation).
I guess I could hack this to assert 1.3 and see what breaks... I'd be pretty nervous about removing this though, even if nothing I try actually breaks.

File: lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp (hunk 1 above)
Comment (jeffnn) [reply lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp]: Thanks!

File: lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp (hunk 2 above)
Comment (jeffnn) [reply lib/DxilPIXPasses/DxilAnnotateWithVirtualRegister.cpp]: Not

File: tools/clang/test/DXC/Passes/PIX/validatorUpgrade.ll (hunk 3 above)
Comment (jeffnn) [reply tools/clang/test/DXC/Passes/PIX/validatorUpgrade.ll]: Didn't think of manually editing in an older valver. New test added.

**Review by bob80905 — APPROVED** [review]
Though I also don't understand the reasoning, the code and tests look correct. Although I would personally put a blurb on top of each test specifying why each test exists. (Like for example, in validatorNoDowngrade, "This test tests that a shader with dxil validator version of 1.8 does not downgrade to 1.4" and for validatorUpgrade "This test tests that a shader with dxil validator version of 1.3 gets upgraded to 1.4 after the annotate with virtual regs pass."

**Comment by jeffnn** [comment]
> Though I also don't understand the reasoning, the code and tests look correct. Although I would personally put a blurb on top of each test specifying why each test exists. (Like for example, in validatorNoDowngrade, "This test tests that a shader with dxil validator version of 1.8 does not downgrade to 1.4" and for validatorUpgrade "This test tests that a shader with dxil validator version of 1.3 gets upgraded to 1.4 after the annotate with virtual regs pass.")

Good idea. Your comment suggestions used, almost verbatim

**Review by tex3d — APPROVED** [review]
The change looks fine, though I still don't know why the validator version was upgraded in the first place.

**Pull request merged**